maths/euclidean_distance.py

```diff
@@ -0,0 +1,22 @@
+import numpy as np
+
+def euclidean_distance(v1, v2) -> double:
```

Comment:
Your typing is not accurate. You cannot directly use double like this.
Read more about typing here https://docs.python.org/3/library/typing.html

How about 

```python
def euclidean_distance(vector_a: np.ndarray, vector_b: np.ndarray ) -> np.float64:
```

If you would also like to support lists and tuples, the typing would get a little complicated.
And you would need to add corresponding doctests as well.
```python
from typing import Union, List, Tuple


vector_tuple = Tuple[Union[float,int]]
vector_list = List[Union[float,int]]
Vector = Union[vector_tuple, vector_list ,np.ndarray]
VectorOut = Union[np.float64, int, float]

def euclidean_distance(vector_a: Vector, vector_b: Vector ) -> VectorOut: 
# convert vector_a and vector_b to np.array and carry out calculations
```

@cclauss Could you please check if this typing is ok or perhaps we can do better?